toolsrc/src/vcpkg/versions.cpp

```diff
@@ -27,4 +46,138 @@ namespace vcpkg::Versions
 
         return hash<string>()(key.port_name) ^ (hash<string>()(key.version.to_string()) >> 1);
     }
+
+    SemanticVersion SemanticVersion::from_string(const std::string& str)
+    {
+        SemanticVersion ret;
+        ret.original_string = str;
+        ret.version_string = str;
+
+        auto build_found = ret.version_string.find('+');
+        if (build_found != std::string::npos)
+        {
+            ret.version_string.resize(build_found);
+        }
+
+        auto prerelease_found = ret.version_string.find('-');
+        if (prerelease_found != std::string::npos)
+        {
+            ret.prerelease_string = ret.version_string.substr(prerelease_found + 1);
+            ret.identifiers = std::move(Strings::split(ret.prerelease_string, '.'));
```

Comment:
```suggestion
            ret.identifiers = Strings::split(ret.prerelease_string, '.');
```
`std::move()` is not needed when the function returns by value.